src/model/application.model.ts

```diff
@@ -1,11 +1,12 @@
-import { entityType, presenterType, dataType, beneficialOwnerTypeType } from "./index";
+import { entityType, presenterType, dataType, beneficialOwnerOtherType, beneficialOwnerTypeType } from "./index";
 
 export const APPLICATION_DATA_KEY = 'roe';
 
 export interface ApplicationData {
     presenter?: presenterType.Presenter;
     entity?: entityType.Entity;
     beneficialOwnerType?: beneficialOwnerTypeType.BeneficialOwnerType;
+    beneficialOwnerOther?: beneficialOwnerOtherType.BeneficialOwnerOther;
 }
 
 export type ApplicationDataType = presenterType.Presenter | entityType.Entity | dataType.Address | beneficialOwnerTypeType.BeneficialOwnerType;
```

Comment:
Don't fully understand what this does, but... do you not need to add 'beneficialOwnerOtherType.BeneficialOwnerOther' on to the end here?

If in fact you do (I've just done this in my PR too) I also wonder how the tests pass without it?

Reply:
prepareData will return data in the 'shape' of ApplicationDataType which I think can be one of those types in the list. But I think at compile time it just looks and sees
`const data: ApplicationDataType = prepareData(req.body, beneficialOwnerTypeType.BeneficialOwnerTypeKeys);`
which satisfies the compiler.
It's only at runtime that the shape of the object returned is defined but I would have thought you'd get an error then but seems to work. So perhaps the ApplicationDataType isn't doing the limiting we hoped, but as all the other types are in that list then this new beneficialOwnerOtherType.BeneficialOwnerOther should be too.

Reply:
👍  worth getting @mouhajer-ch 's thoughts on this one too.